### DISCUSSION TITLE
[Android] Preload and cache videos using Media Element

### DISCUSSION THREAD

**Discussion by EdwardMcFlurry** [CommunityToolkit/Maui#1144] [Q&A]
Hi

I’m asking for any assistance with regards to preloading and caching the videos in NET Maui particularly on Android. On iOS, it takes literally 2-3 seconds to load and instantly play the Video but takes roughly 15-20 seconds on Android.

Is this an issue with Exoplayer or is there any code that I should add in order to make my App more performant for Users?

The videos are on Azure Blob Storage!

**Comment by ne0rrmatrix** [comment]
When  I load video's on Android using media element it loads in a a few seconds. Same on IOS and Windows. There is no noticeable difference between load times for me at all.

**Reply by EdwardMcFlurry** [reply]
What about the Caching Mechanism, do you if one can implement such?

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
